sdk/core/azure-core/inc/azure/core/uuid.hpp

```diff
@@ -54,6 +58,46 @@ namespace Azure { namespace Core {
      * @brief Construct a Uuid from an existing UUID represented as an array of bytes.
      * @details Creates a Uuid from a UUID created in an external scope.
      */
-    static Uuid CreateFromArray(std::array<uint8_t, UuidSize> const& uuid);
+    static constexpr Uuid CreateFromArray(ValueArray const& uuid) { return Uuid{uuid}; }
+
+    /**
+     * @brief Construct a Uuid by parsing its representation.
+     * @param s a string in `8-4-4-4-12` hex characters format.
+     * @throw `std::invalid_argument` if \p s cannot be parsed.
+     */
+    static Uuid Parse(std::string const& s);
+
+    constexpr bool operator==(Uuid const& other) const
+    {
+      // std::array::operator==() is not a constexpr until C++20
```

Comment:
if (m_uuid.size() != other.m_uuid.size())
   return false;

Reply:
I don't see how it is realistically possible - std::array has a constexpr size as it's template parameter.